[choonkit-nus] iP

choonkit-nus opened the pull request.

choonkit-nus committed:
Rename bot, Greet, Exit

choonkit-nus committed:
Echo bot

choonkit-nus committed:
Edit bot to Add + List

choonkit-nus committed:
Add Task class and mark feature

choonkit-nus committed:
Add todo, event, deadline feature with Task class inheritance

irving11119 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,97 @@
+
+public class Duke {
+    public static void main(String[] args) {
+        String logo = "       _       _        \n"
+                + "      | |     | |       \n"
+                + "      | | __ _| | _____ \n"
+                + "  _   | |/ _` | |/ / _ \\\n"
+                + " | |__| | (_| |   <  __/\n"
+                + "  \\____/ \\__,_|_|\\_\\___|\n";
```
Perhaps you could make this a constant instead.

irving11119 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,97 @@
+        System.out.println(logo);
+        System.out.println("Hello! I'm Jake");
+        System.out.println("What can I do for you?");
+        Scanner readIn = new Scanner(System.in);
```
It's better to use nouns as class names rather than verbs. So perhaps you could name this `Scanner inputReader`

irving11119 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,97 @@
+
+            // Bot Event task feature (use: event [desc] /from [start] /to [end])
+            if (text[0].equalsIgnoreCase("event")) {
+                int startIndex = userInput.indexOf("/from ");
+                int endIndex = userInput.indexOf("/to ");
+                String startTime = userInput.substring(startIndex + 6, endIndex);
+                String endTime = userInput.substring(endIndex + 4);
+                int descriptionIndex = userInput.indexOf(text[1]);
+                String description = userInput.substring(descriptionIndex, startIndex);
+                Task t = new Event(description, startTime, endTime);
+                botMemory.add(t);
+                System.out.println("Added the following task:");
+                System.out.println(t.getDescription());
+                System.out.println("Now there are " + Task.getTaskCount() + " tasks in the botMemory list!");
```
Perhaps you could leave spaces in between statements to make your code more readable.

For example, you could add a line between your index variables and time variables.

irving11119 reviewed (commented):
Overall your code is quite readable and it follows the standards well. Just remember to leave appropriate blank lines when you have multiple lines grouped together.

choonkit-nus committed:
Tweak code quality

Based on comments of PR review,
Set logo to be a constant variable
Edited readIn to inputReader (to use noun instead of verb for name convention of class)

choonkit-nus committed:
Add DukeException class to represent exceptions specific to Duke

choonkit-nus committed:
Add error handling for incorrect input exceptions

choonkit-nus committed:
Merge branch 'branch-Level-5'

choonkit-nus committed:
Add delete task feature

choonkit-nus committed:
Add save feature for botMemory to data/botMemory.txt file

choonkit-nus committed:
Merge branch 'branch-Level-6'

choonkit-nus committed:
Merge branch 'branch-Level-7'

choonkit-nus committed:
Add find task feature

choonkit-nus committed:
Add JavaDoc comments

choonkit-nus committed:
Add user guide

choonkit-nus committed:
Revert "Add user guide"

This reverts commit be1a89e186ed945e6d5b1bca4e0c5bbbb44c85f3.

choonkit-nus committed:
Add user guide

choonkit-nus committed:
Release v1.0

choonkit-nus committed:
Bug fixes for botMemory file creation

If the botMemory file does not exist, it then creates the "data" directory and "botMemory.txt" file

choonkit-nus committed:
Release v1.1
